### PR TITLE
Add NER and classification scripts

### PR DESCRIPTION
**Pull request by Ciroye**

### CODE REVIEW

**Comment by galv** [comment]
Before merging, can you add an Apache 2.0 license header to all python files?

This one: https://gist.github.com/Rypac/7349957f95f0713143fda2f3d3bc10c8

<OWNER> should be MLCommons in your case (I am double-checking to confirm this)

File: galvasr2/ner_pipeline_transcript.py
Comment (galv) [inline galvasr2/ner_pipeline_transcript.py:108]: Ah, autoformatting, very nice.

File: galvasr2/ner_pipeline_transcript.py
Comment (galv) [inline galvasr2/ner_pipeline_transcript.py:60]: As an FYI, if your set of keys isn't known in advanced collections.defaultdict(list) is a powerful tool you can use. I don't recommend changing this code, though. It is helpful to have the tags visible for understanding.

File: galvasr2/zero_shot_classification_pipeline.py
Comment (galv) [inline galvasr2/zero_shot_classification_pipeline.py:27]: Again, DEFINE_integer

File: galvasr2/zero_shot_classification_pipeline.py
Comment (galv) [inline galvasr2/zero_shot_classification_pipeline.py:65]: change to "health"

File: galvasr2/zero_shot_classification_pipeline.py
Comment (galv) [inline galvasr2/zero_shot_classification_pipeline.py:67]: change to "finance"

File: galvasr2/zero_shot_classification_pipeline.py
Comment (galv) [inline galvasr2/zero_shot_classification_pipeline.py:73]: It looks like you are truncating the string you are classifying. How come? Do you get out of memory errors if the string is too large? If so, can you comment that?

File: galvasr2/zero_shot_classification_pipeline.py
Comment (galv) [inline galvasr2/zero_shot_classification_pipeline.py:78]: Running at a larger batch size is a straightforward way to try to speed things up. Does zsl.predict automatically use a GPU?

**Review by galv — COMMENTED** [review]
I will hit the approve button after you add the license headers to the ".py" files. Are the notebooks the same as the scripts?

File: galvasr2/ner_pipeline_transcript.py
Comment (Ciroye) [reply galvasr2/ner_pipeline_transcript.py:20]: Exactly, well with a little more exploration about the data, the idea is that someone new could follow these notebooks to understand all the process

File: galvasr2/ner_pipeline_transcript.py
Comment (Ciroye) [reply galvasr2/ner_pipeline_transcript.py:27]: Sorry, totally my fault, fixed it

File: galvasr2/ner_pipeline_transcript.py
Comment (Ciroye) [reply galvasr2/ner_pipeline_transcript.py:49]: Oki doki

File: galvasr2/ner_pipeline_transcript.py
Comment (Ciroye) [reply galvasr2/ner_pipeline_transcript.py:92]: Around 38 hours for all the data, more or less (a weekend)

File: galvasr2/zero_shot_classification_pipeline.py
Comment (Ciroye) [reply galvasr2/zero_shot_classification_pipeline.py:27]: Sorry again =)

File: galvasr2/zero_shot_classification_pipeline.py
Comment (Ciroye) [reply galvasr2/zero_shot_classification_pipeline.py:65]: Done

File: galvasr2/zero_shot_classification_pipeline.py
Comment (Ciroye) [reply galvasr2/zero_shot_classification_pipeline.py:67]: Done

File: galvasr2/zero_shot_classification_pipeline.py
Comment (Ciroye) [reply galvasr2/zero_shot_classification_pipeline.py:73]: This is a little tricky, there some transcriptions that are very large, the problem is not about memory errors or something similar, is about time, I think that with the batch size that you comment I could fix that

File: galvasr2/zero_shot_classification_pipeline.py
Comment (Ciroye) [reply galvasr2/zero_shot_classification_pipeline.py:78]: You're totally right

File: galvasr2/zero_shot_classification_pipeline.py
Comment (Ciroye) [reply galvasr2/zero_shot_classification_pipeline.py:73]: Another reason, was because in theory, if you read, I don't know, 30% of a document, you can infer with high accuracy the most accurate category

Comment (Ciroye) [reply galvasr2/zero_shot_classification_pipeline.py:73]: "Prediction Time and Batch Size
The predict method can accept a large list of documents. Documents are automatically split into batches based on the batch_size parameter, which can be increased to speed up predictions.

Note also that the predict method of ZeroShotClassifier generates a separate NLI prediction for each label included in the labels parameter. As len(labels) and the number of documents fed to predict increases, the prediction time will also increase. You can speed up predictions by increasing the batch_size. The default batch_size is currently set conservatively at 8:

- Predicting 800 topics for a single document takes ~26 seconds on a TITAN V GPU using batch_size=1
- Predicting 800 topics for a single document takes less than 2 seconds on a TITAN V GPU using batch_size=64¶
""
From the official documentation 

You were completely right, sorry

Comment (galv) [reply galvasr2/zero_shot_classification_pipeline.py:73]: It looks like increasing the batch_size is not a magic bullet. I guess you cannot increase it beyond the number of zero shot classification categories we use, if I am understanding correctly. I'm guessing this is a transformer model. Still, it may be worthwhile to set batch_size=len(labels). It you can evaluate the speedup on a small amount of data, that would be great. If it is too hard, I don't mind if you just leave this alone. 38 hours on a rarely run job is not horrible.

File: galvasr2/ner_pipeline_transcript.py
Comment (Ciroye) [reply galvasr2/ner_pipeline_transcript.py:60]: Good suggestion, thanks

File: galvasr2/ner_pipeline_transcript.py
Comment (Ciroye) [reply galvasr2/ner_pipeline_transcript.py:108]: haha black to the rescue

**Review by galv — APPROVED** [review]

**Pull request merged**
